### PR TITLE
Retire DoH for UD and ENS

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/22149
- Removed DNS over HTTPS code. Users having this turned on will have pref reset to default.
- Inlined DecentralizedDns feature
- Cleaned usage of DECENTRALIZED_DNS_ENABLED buildflag

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by iefremov — APPROVED** [review]
browser/net/* LGTM, haven't checked anything else

File: net/sources.gni [hunk 1]
```diff
@@ -3,12 +3,8 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import("//brave/components/decentralized_dns/buildflags/buildflags.gni")
-
 brave_net_sources = [
   "//brave/net/decentralized_dns/constants.h",
-  "//brave/net/dns/brave_resolve_context.cc",
-  "//brave/net/dns/brave_resolve_context.h",
```
Comment (goodov) [inline net/sources.gni:11]: I think we can drop `brave_dns_friend` now

File: browser/decentralized_dns/test/BUILD.gn [hunk 2]
```diff
@@ -36,8 +33,6 @@ source_set("unit_tests") {
     "//brave/browser/decentralized_dns/test/decentralized_dns_navigation_throttle_unittest.cc",
     "//brave/browser/decentralized_dns/test/utils_unittest.cc",
     "//brave/browser/net/decentralized_dns_network_delegate_helper_unittest.cc",
-    "//brave/net/dns/brave_resolve_context_unittest.cc",
-    "//brave/net/dns/dns_transaction_unittest.cc",
```
Comment (goodov) [inline browser/decentralized_dns/test/BUILD.gn:40]: we can drop `brave_dns_friend` now.

**Review by yrliou — APPROVED** [review]
++

File: browser/decentralized_dns/test/BUILD.gn (hunk 2 above)
Comment (supermassive) [reply browser/decentralized_dns/test/BUILD.gn:40]: done in https://github.com/brave/brave-core/pull/13354/commits/ac2744997363aabb5f8662f1dc05c053c3aa1a96

**Comment by kjozwiak** [comment]
Verification on Nightly as done via https://github.com/brave/brave-core/pull/13427#issuecomment-1137240836.

**Pull request merged**